deeplake/util/version_control.py

```diff
@@ -548,6 +580,60 @@ def replace_head(storage, version_state, reset_commit_id):
     return new_commit_id
 
 
+def find_commits(storage) -> List:
```

Comment:
s3 listing objects is pretty slow, commits should instead be found by traversing starting from firstdb commit, which points to the next locations to check

Reply:
this is also to account for commit info files that could be missing. otherwise won't firstdb commit be another point of failure.